### PR TITLE
Use JSONL diagnostics file

### PR DESCRIPTION
**Pull request by W4ngatang**
A few quality of life improvements for "linguistic" diagnostic:

- use JSONL with category labels
- track MCC for all examples
- track accuracy over all examples
- remove "*__@@UNK@@" from label scorers

### CODE REVIEW

**Comment by pruksmhc** [comment]
Don't forget to run black1

File: jiant/evaluate.py [hunk 1]
```diff
@@ -88,10 +88,8 @@ def evaluate(
                 batch = move_to_device(batch, cuda_device)
                 out = model.forward(task, batch, predict=True)
 
-            # We don't want diagnostic tasks to affect the micro and macro average.
```
Comment (sleepinyourhat) [inline jiant/evaluate.py]: Why are we deleting this? Was this never actually used in the averages?

File: jiant/evaluate.py [hunk 2]
```diff
@@ -120,9 +118,12 @@ def evaluate(
...
             all_metrics["%s_%s" % (task.name, name)] = value
-        all_metrics["micro_avg"] += all_metrics[task.val_metric] * n_examples
-        all_metrics["macro_avg"] += all_metrics[task.val_metric]
-        n_examples_overall += n_examples
+
+        # We don't want diagnostic tasks to affect the micro and macro average.
+        if not isinstance(task, GLUEDiagnosticTask):
+            all_metrics["micro_avg"] += all_metrics[task.val_metric] * n_examples
+            all_metrics["macro_avg"] += all_metrics[task.val_metric]
+            n_examples_overall += n_examples
```
Comment (sleepinyourhat) [inline jiant/evaluate.py]: See if you can add names/comments to more clearly distinguish these two vars.

File: jiant/tasks/tasks.py [hunk 3]
```diff
@@ -1193,9 +1193,9 @@ def load_data(self):
 
 
 @register_task("glue-diagnostic", rel_path="MNLI/", n_classes=3)
-@register_task("superglue-diagnostic", rel_path="RTE/", n_classes=2)
+@register_task("glue-diagnostic-binary", rel_path="RTE/", n_classes=2)
```
Comment (sleepinyourhat) [inline jiant/tasks/tasks.py]: Add a comment explaining how/why this differs from the SuperGLUE version.

File: jiant/tasks/tasks.py [hunk 4]
```diff
@@ -1387,9 +1392,120 @@ def collect_metrics(ix_to_tag_dict, tag_group):
...
         collect_metrics(self.ix_to_logic_dic, "logic")
         collect_metrics(self.ix_to_knowledge_dic, "knowledge")
+        collected_metrics["all_mcc"] = self._scorer_all_mcc.get_metric(reset)
+        collected_metrics["accuracy"] = self._scorer_all_acc.get_metric(reset)
         return collected_metrics
 
 
+@register_task("superglue-diagnostic", rel_path="RTE/")
+class SuperGLUEDiagnosticTask(GLUEDiagnosticTask):
+    """ Task class for SuperGLUE diagnostic data """
```
Comment (sleepinyourhat) [inline jiant/tasks/tasks.py]: Use a more specific name to separate this from WinoGender.

File: jiant/evaluate.py (hunk 1 above)
Comment (W4ngatang) [reply jiant/evaluate.py]: We just don't use `n_examples` for diagnostic tasks later.

**Comment by W4ngatang** [comment]
TODO: figure out broad coverage diagnostic prediction file naming convention with Amanpreet

**Pull request merged**